frontend/src/components/Adminochka/Statistics/StudentsTable.jsx

```diff
@@ -14,20 +14,28 @@ import {
   ORDER,
   TASKS_NUMBER,
   SELECTED_CONTEST,
+  PAGE,
 } from "components/Adminochka/Statistics/Utils";
 import StudentsTableContent from "components/Adminochka/Statistics/StudentsTableContent";
 
 const StudentsTable = ({
   studentsList,
+  contests,
   selectedFilters,
   setFilter,
   getStudents,
 }) => {
   const [loading, setLoading] = useState({});
   const isContestSelected = selectedFilters[SELECTED_CONTEST];
 
-  //TODO заменить хардкоженое число задач в контесте на полученное откуда-то (возможно из ресурса отдающего список контестов)
-  const numberOfTasks = ["0", "1", "2"];
+  let numberOfTasks = [];
+  contests.forEach(contest => {
+    if (contest.id === selectedFilters[SELECTED_CONTEST]) {
+      for (let i = 0; i < contest.numberOfTasks + 1; i++) {
+        numberOfTasks.push(i.toString());
+      }
+    }
+  });
```

Comment:
```suggestion
const numberOfTasks = contests.filter((contest) => contest.id === selectedFilters[SELECTED_CONTEST]).map((contest) => [...Array(contest.numberOfTasks).keys()])
```

Можно наверн еще так, но тогда численные значения будут, надо еще добавлять map, чтобы строкой сделать

Reply:
я вот так тоже смотрел, но мне показалось слишком много тут понатыкано, ну ладно, исправлю)

Reply:
только так как ты хочешь не заработает, filter возвращает значение, то есть в данном случае объект, от которого map нельзя

Reply:
твой вариант все же не работает, ты map применяешь к [{...}] и в итоге только одно значение получается, а не сколько пришло в numberOfTasks

Reply:
можно построить массив из ...Array(нулевой элемент полученного массива), но там пока ответ с бэка не пришел бывает и undefined

Reply:
ну и там фишка что к значению внутри ...Array() нужно прибавить 1, чтобы корректно отображались возможные задачи в селекте

Reply:
Короче оставляй, как есть